### PR TITLE
Model validations

### PR DESCRIPTION
**Pull request by kiboma2021**
Use validations for models.
Write unit tests for models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
 
+  validates :name, presence: true
+
```
Comment (BenMukebo) [inline app/models/user.rb]: - [ ] According to the project requirements you should add the following validations for the User model: PostsCounter must be an integer greater than or equal to zero. Kindly add this it should be something like this:
`validates_numericality_of :posts_counter, only_integer: true, greater_than_or_equal: 0`

File: app/models/post.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :comments, dependent: :delete_all
   has_many :likes, dependent: :delete_all
 
+  validates :text, presence: true, length: { maximum: 250, minimum: 1 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
   after_create :update_posts_count
```
Comment (BenMukebo) [inline app/models/post.rb]: - [ ] According to the project requirements you should add the following validations for the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
Kindly add this ✍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @kiboma2021,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Professional README ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] After adding the missing validation 👇🏻 Kindly make sure to Add unit specs for all of your models' methods and validations. 💯 👍🏻 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (kiboma2021) [reply app/models/user.rb]: @BenMukebo , This has now been implemented

File: app/models/post.rb (hunk 2 above)
Comment (kiboma2021) [reply app/models/post.rb]: @BenMukebo , this has now been implemented

**Review by Deepakdanger — APPROVED** [review]
Hi @kiboma2021 , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**